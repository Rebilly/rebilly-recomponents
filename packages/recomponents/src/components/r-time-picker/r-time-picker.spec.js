import {shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RTimePicker from './r-time-picker.vue';

describe('r-time-picker.vue', () => {
    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RTimePicker, {
            mocks: {
                $t: () => {
                },
            },
            propsData: {
                id: 'id',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RTimePicker, {
            mocks: {
                $t: () => {
                },
            },
            propsData: {
                id: 'id',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
