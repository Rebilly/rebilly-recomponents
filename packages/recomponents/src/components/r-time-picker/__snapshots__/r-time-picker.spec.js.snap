// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-time-picker.vue should render Wrapper and match snapshot 1`] = `
<div class="r-field r-time-picker__wrapper">
  <!---->
  <div class="r-time-picker">
    <div class="r-time-picker__input">
      <r-select-stub asyncconfig="[object Object]" allowempty="true" blockkeys="" clearonselect="true" closeonselect="true" customlabel="default(option, label) {
        if (isEmpty(option)) {
          return '';
        }

        return label ? option[label] : option;
      }" id="id-hours" internalsearch="true" limit="99999" limittext="function (count) {
        return this.$t('more', {
          count
        });
      }" maxheight="300" name="" opendirection="" options="00,01,02,03,04,05,06,07,08,09,10,11,12,13,14,15,16,17,18,19,20,21,22,23" optionheight="40" optionslimit="1000" searchable="true" shownooptions="true" shownoresults="true" showpointer="true" tabindex="0" tagposition="top" value=""></r-select-stub>
    </div>
    <div class="r-time-picker__divider"><span>:</span></div>
    <div class="r-time-picker__input">
      <r-select-stub asyncconfig="[object Object]" allowempty="true" blockkeys="" clearonselect="true" closeonselect="true" customlabel="default(option, label) {
        if (isEmpty(option)) {
          return '';
        }

        return label ? option[label] : option;
      }" id="id-minutes" internalsearch="true" limit="99999" limittext="function (count) {
        return this.$t('more', {
          count
        });
      }" maxheight="300" name="" opendirection="" options="00,01,02,03,04,05,06,07,08,09,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59" optionheight="40" optionslimit="1000" searchable="true" shownooptions="true" shownoresults="true" showpointer="true" tabindex="0" tagposition="top" value=""></r-select-stub>
    </div>
  </div>
  <!---->
</div>
`;

exports[`r-time-picker.vue should render via SSR and match snapshot 1`] = `
<div class="r-field r-time-picker__wrapper">
  <!---->
  <div class="r-time-picker">
    <div class="r-time-picker__input">
      <div>
        <!---->
        <div tabindex="-1" role="combobox" aria-owns="listbox-id-hours" class="r-select r-field r-select--allow-empty">
          <div class="r-select__select"></div>
          <div class="r-select__tags">
            <div class="r-select__tags-wrap" style="display:none;"></div>
            <!---->
            <div name="r-select__loading" class="r-select__spinner" style="display:none;"></div> <input name="" id="id-hours" type="text" autocomplete="off" spellcheck="false" tabindex="0" aria-controls="listbox-id-hours" value="" class="r-select__input" style="width:0;position:absolute;padding:0;">
            <!----> <span class="r-select__placeholder">

                </span>
          </div>
          <div tabindex="-1" name="r-select" class="r-select__content-wrapper" style="max-height:300px;display:none;">
            <ul role="listbox" id="listbox-id-hours" class="r-select__content" style="display:inline-block;">
              <!---->
              <!---->
              <!---->
              <li id="id-hours-0" role="00" class="r-select__element"><span class="r-select__option r-select__option--highlight"><span>00</span></span></li>
              <li id="id-hours-1" role="01" class="r-select__element"><span class="r-select__option"><span>01</span></span></li>
              <li id="id-hours-2" role="02" class="r-select__element"><span class="r-select__option"><span>02</span></span></li>
              <li id="id-hours-3" role="03" class="r-select__element"><span class="r-select__option"><span>03</span></span></li>
              <li id="id-hours-4" role="04" class="r-select__element"><span class="r-select__option"><span>04</span></span></li>
              <li id="id-hours-5" role="05" class="r-select__element"><span class="r-select__option"><span>05</span></span></li>
              <li id="id-hours-6" role="06" class="r-select__element"><span class="r-select__option"><span>06</span></span></li>
              <li id="id-hours-7" role="07" class="r-select__element"><span class="r-select__option"><span>07</span></span></li>
              <li id="id-hours-8" role="08" class="r-select__element"><span class="r-select__option"><span>08</span></span></li>
              <li id="id-hours-9" role="09" class="r-select__element"><span class="r-select__option"><span>09</span></span></li>
              <li id="id-hours-10" role="10" class="r-select__element"><span class="r-select__option"><span>10</span></span></li>
              <li id="id-hours-11" role="11" class="r-select__element"><span class="r-select__option"><span>11</span></span></li>
              <li id="id-hours-12" role="12" class="r-select__element"><span class="r-select__option"><span>12</span></span></li>
              <li id="id-hours-13" role="13" class="r-select__element"><span class="r-select__option"><span>13</span></span></li>
              <li id="id-hours-14" role="14" class="r-select__element"><span class="r-select__option"><span>14</span></span></li>
              <li id="id-hours-15" role="15" class="r-select__element"><span class="r-select__option"><span>15</span></span></li>
              <li id="id-hours-16" role="16" class="r-select__element"><span class="r-select__option"><span>16</span></span></li>
              <li id="id-hours-17" role="17" class="r-select__element"><span class="r-select__option"><span>17</span></span></li>
              <li id="id-hours-18" role="18" class="r-select__element"><span class="r-select__option"><span>18</span></span></li>
              <li id="id-hours-19" role="19" class="r-select__element"><span class="r-select__option"><span>19</span></span></li>
              <li id="id-hours-20" role="20" class="r-select__element"><span class="r-select__option"><span>20</span></span></li>
              <li id="id-hours-21" role="21" class="r-select__element"><span class="r-select__option"><span>21</span></span></li>
              <li id="id-hours-22" role="22" class="r-select__element"><span class="r-select__option"><span>22</span></span></li>
              <li id="id-hours-23" role="23" class="r-select__element"><span class="r-select__option"><span>23</span></span></li>
              <li style="display:none;"><span class="r-select__option">
                            </span></li>
              <li style="display:none;"><span class="r-select__option"></span></li>
              <!---->
            </ul>
          </div>
        </div>
        <!---->
      </div>
    </div>
    <div class="r-time-picker__divider"><span>:</span></div>
    <div class="r-time-picker__input">
      <div>
        <!---->
        <div tabindex="-1" role="combobox" aria-owns="listbox-id-minutes" class="r-select r-field r-select--allow-empty">
          <div class="r-select__select"></div>
          <div class="r-select__tags">
            <div class="r-select__tags-wrap" style="display:none;"></div>
            <!---->
            <div name="r-select__loading" class="r-select__spinner" style="display:none;"></div> <input name="" id="id-minutes" type="text" autocomplete="off" spellcheck="false" tabindex="0" aria-controls="listbox-id-minutes" value="" class="r-select__input" style="width:0;position:absolute;padding:0;">
            <!----> <span class="r-select__placeholder">

                </span>
          </div>
          <div tabindex="-1" name="r-select" class="r-select__content-wrapper" style="max-height:300px;display:none;">
            <ul role="listbox" id="listbox-id-minutes" class="r-select__content" style="display:inline-block;">
              <!---->
              <!---->
              <!---->
              <li id="id-minutes-0" role="00" class="r-select__element"><span class="r-select__option r-select__option--highlight"><span>00</span></span></li>
              <li id="id-minutes-1" role="01" class="r-select__element"><span class="r-select__option"><span>01</span></span></li>
              <li id="id-minutes-2" role="02" class="r-select__element"><span class="r-select__option"><span>02</span></span></li>
              <li id="id-minutes-3" role="03" class="r-select__element"><span class="r-select__option"><span>03</span></span></li>
              <li id="id-minutes-4" role="04" class="r-select__element"><span class="r-select__option"><span>04</span></span></li>
              <li id="id-minutes-5" role="05" class="r-select__element"><span class="r-select__option"><span>05</span></span></li>
              <li id="id-minutes-6" role="06" class="r-select__element"><span class="r-select__option"><span>06</span></span></li>
              <li id="id-minutes-7" role="07" class="r-select__element"><span class="r-select__option"><span>07</span></span></li>
              <li id="id-minutes-8" role="08" class="r-select__element"><span class="r-select__option"><span>08</span></span></li>
              <li id="id-minutes-9" role="09" class="r-select__element"><span class="r-select__option"><span>09</span></span></li>
              <li id="id-minutes-10" role="10" class="r-select__element"><span class="r-select__option"><span>10</span></span></li>
              <li id="id-minutes-11" role="11" class="r-select__element"><span class="r-select__option"><span>11</span></span></li>
              <li id="id-minutes-12" role="12" class="r-select__element"><span class="r-select__option"><span>12</span></span></li>
              <li id="id-minutes-13" role="13" class="r-select__element"><span class="r-select__option"><span>13</span></span></li>
              <li id="id-minutes-14" role="14" class="r-select__element"><span class="r-select__option"><span>14</span></span></li>
              <li id="id-minutes-15" role="15" class="r-select__element"><span class="r-select__option"><span>15</span></span></li>
              <li id="id-minutes-16" role="16" class="r-select__element"><span class="r-select__option"><span>16</span></span></li>
              <li id="id-minutes-17" role="17" class="r-select__element"><span class="r-select__option"><span>17</span></span></li>
              <li id="id-minutes-18" role="18" class="r-select__element"><span class="r-select__option"><span>18</span></span></li>
              <li id="id-minutes-19" role="19" class="r-select__element"><span class="r-select__option"><span>19</span></span></li>
              <li id="id-minutes-20" role="20" class="r-select__element"><span class="r-select__option"><span>20</span></span></li>
              <li id="id-minutes-21" role="21" class="r-select__element"><span class="r-select__option"><span>21</span></span></li>
              <li id="id-minutes-22" role="22" class="r-select__element"><span class="r-select__option"><span>22</span></span></li>
              <li id="id-minutes-23" role="23" class="r-select__element"><span class="r-select__option"><span>23</span></span></li>
              <li id="id-minutes-24" role="24" class="r-select__element"><span class="r-select__option"><span>24</span></span></li>
              <li id="id-minutes-25" role="25" class="r-select__element"><span class="r-select__option"><span>25</span></span></li>
              <li id="id-minutes-26" role="26" class="r-select__element"><span class="r-select__option"><span>26</span></span></li>
              <li id="id-minutes-27" role="27" class="r-select__element"><span class="r-select__option"><span>27</span></span></li>
              <li id="id-minutes-28" role="28" class="r-select__element"><span class="r-select__option"><span>28</span></span></li>
              <li id="id-minutes-29" role="29" class="r-select__element"><span class="r-select__option"><span>29</span></span></li>
              <li id="id-minutes-30" role="30" class="r-select__element"><span class="r-select__option"><span>30</span></span></li>
              <li id="id-minutes-31" role="31" class="r-select__element"><span class="r-select__option"><span>31</span></span></li>
              <li id="id-minutes-32" role="32" class="r-select__element"><span class="r-select__option"><span>32</span></span></li>
              <li id="id-minutes-33" role="33" class="r-select__element"><span class="r-select__option"><span>33</span></span></li>
              <li id="id-minutes-34" role="34" class="r-select__element"><span class="r-select__option"><span>34</span></span></li>
              <li id="id-minutes-35" role="35" class="r-select__element"><span class="r-select__option"><span>35</span></span></li>
              <li id="id-minutes-36" role="36" class="r-select__element"><span class="r-select__option"><span>36</span></span></li>
              <li id="id-minutes-37" role="37" class="r-select__element"><span class="r-select__option"><span>37</span></span></li>
              <li id="id-minutes-38" role="38" class="r-select__element"><span class="r-select__option"><span>38</span></span></li>
              <li id="id-minutes-39" role="39" class="r-select__element"><span class="r-select__option"><span>39</span></span></li>
              <li id="id-minutes-40" role="40" class="r-select__element"><span class="r-select__option"><span>40</span></span></li>
              <li id="id-minutes-41" role="41" class="r-select__element"><span class="r-select__option"><span>41</span></span></li>
              <li id="id-minutes-42" role="42" class="r-select__element"><span class="r-select__option"><span>42</span></span></li>
              <li id="id-minutes-43" role="43" class="r-select__element"><span class="r-select__option"><span>43</span></span></li>
              <li id="id-minutes-44" role="44" class="r-select__element"><span class="r-select__option"><span>44</span></span></li>
              <li id="id-minutes-45" role="45" class="r-select__element"><span class="r-select__option"><span>45</span></span></li>
              <li id="id-minutes-46" role="46" class="r-select__element"><span class="r-select__option"><span>46</span></span></li>
              <li id="id-minutes-47" role="47" class="r-select__element"><span class="r-select__option"><span>47</span></span></li>
              <li id="id-minutes-48" role="48" class="r-select__element"><span class="r-select__option"><span>48</span></span></li>
              <li id="id-minutes-49" role="49" class="r-select__element"><span class="r-select__option"><span>49</span></span></li>
              <li id="id-minutes-50" role="50" class="r-select__element"><span class="r-select__option"><span>50</span></span></li>
              <li id="id-minutes-51" role="51" class="r-select__element"><span class="r-select__option"><span>51</span></span></li>
              <li id="id-minutes-52" role="52" class="r-select__element"><span class="r-select__option"><span>52</span></span></li>
              <li id="id-minutes-53" role="53" class="r-select__element"><span class="r-select__option"><span>53</span></span></li>
              <li id="id-minutes-54" role="54" class="r-select__element"><span class="r-select__option"><span>54</span></span></li>
              <li id="id-minutes-55" role="55" class="r-select__element"><span class="r-select__option"><span>55</span></span></li>
              <li id="id-minutes-56" role="56" class="r-select__element"><span class="r-select__option"><span>56</span></span></li>
              <li id="id-minutes-57" role="57" class="r-select__element"><span class="r-select__option"><span>57</span></span></li>
              <li id="id-minutes-58" role="58" class="r-select__element"><span class="r-select__option"><span>58</span></span></li>
              <li id="id-minutes-59" role="59" class="r-select__element"><span class="r-select__option"><span>59</span></span></li>
              <li style="display:none;"><span class="r-select__option">
                            </span></li>
              <li style="display:none;"><span class="r-select__option"></span></li>
              <!---->
            </ul>
          </div>
        </div>
        <!---->
      </div>
    </div>
  </div>
  <!---->
</div>
`;
